shap/plots/_bar.py

```diff
@@ -54,17 +55,35 @@ def bar(shap_values, max_display=10, order=Explanation.abs, clustering=None, clu
         cohorts = {"": shap_values}
     elif isinstance(shap_values, Cohorts):
         cohorts = shap_values.cohorts
+    elif isinstance(shap_values, dict):
+        cohorts = shap_values
     else:
-        assert isinstance(shap_values, dict), "You must pass an Explanation object, Cohorts object, or dictionary to bar plot!"
+        emsg = (
+            "The shap_values argument must be an Explanation object, Cohorts "
+            "object, or dictionary of Explanation objects!"
+        )
+        raise TypeError(emsg)
```

Comment:
Same comment as other PR - suggest remove the redundant intermediate variable `emsg`

Reply:
https://beta.ruff.rs/docs/rules/raw-string-in-exception/#rule-selection

The reason why I choose to do so is explained here ^

Reply:
Thanks for the link. I didn't find the given argument persuasive: it feels like an ugly hack to try to modify the builtin traceback formatting.

The "flake8-errmsg" plugin doesn't seem widely used, and other libs (e.g. pandas) seem to implement exceptions directly rather than with this workaround. Have you seen this pattern for exceptions applied elsewhere in other projects?

Reply:
> Have you seen this pattern for exceptions applied elsewhere in other projects?

Not really.

What's the argument for _not_ using this pattern, other than the fact that it saves us one line of code?

Reply:
> What's the argument for not using this pattern, other than the fact that it saves us one line of code?

Yeah it's simply that, just saving a line of code and keep things minimalistic and simple where possible. Maybe @dsgibbons will cast a tie-breaking vote.

For now let's "agree to disagree" and move on as it's only a style issue. I'll approve the PR :)

Reply:
Thanks :)